### ISSUE TITLE
ftp.halifax.rwth-aachen.de blocked my provider

### ISSUE THREAD

**Issue by ihatethecloud** [jenkins-infra/helpdesk#3956]
### Service(s)

Update center

### Summary

It seems https://ftp.halifax.rwth-aachen.de is blocking my provider(RCS-RDS - ASN8708) in Romania. I did try https://ftp.belnet.be and it works fine but I have no idea how to force the use of that. Is there an option for it ?

**Comment by ihatethecloud** [comment]
Sent an email to hrwth-aachen.de and I got the following reply:
```your provider is unable to respond to abuse mails. Please ask them to
get into touch with us, so that we can find a solution for the existing
issues. In the mean time, customers of RCS RDS remain blocked.

I strongly suggest to switch to another provider.

Best regards
```

Which sounds very unreasonable.

**Comment by dduportal** [comment]
Hi @C-Otto  , it looks like another case being a consequence of #3930, like #3935.

Is there anything that could be done to help in this case?

**Comment by dduportal** [comment]
> Sent an email to hrwth-aachen.de and I got the following reply:
> 
> ```
> your provider is unable to respond to abuse mails. Please ask them to
> get into touch with us, so that we can find a solution for the existing
> issues. In the mean time, customers of RCS RDS remain blocked.
> 
> I strongly suggest to switch to another provider.
> 
> Best regards
> ```
> 
> Which sounds very unreasonable.

Sorry for this @ihatethecloud but it is not as simple: the Jenkins project only lives from sponsorchips and the Aachen university is providing us a mirror for free. In a world where bandwidth costs a lot, it help having the project to be sustainable.

A long term solution would be to find a mirror provider (requires an HTTP + Rsync server with 750 Gb hard drive and of course good network) to provide a local copy in your country or area of the world.
That would ensure Aachen mirror is not used in this area and would solve your problem.

Do you know if your organisation (or any around you) which could help on this matter? Eventually your ISP?

**Comment by C-Otto** [comment]
Hi, this is not related to the linked issues. In fact, several networks of this provider have been blocked since late 2016.

These networks remain blocked because the provider is unable to respond to abuse mails in time (we have been ignored more than once). In cases where we were able to discuss issues, they refused to work towards solutions and instead resorted to threats and pointless reminder mails (while ignoring our responses). From our point of view, it's more important to provide a healthy service to users of other providers than suffering from abusive RCS-RDS customers affecting the whole server.

While we're at it, several (very) large networks of Chine Mobile are also blocked as their abuse address is broken (and because we continue to face abuse from their networks).

**Comment by C-Otto** [comment]
We'd be happy to unblock the networks in question after resolving the underlying technical and communication issues. I already provided the ticket number to @ihatethecloud and outlined the open questions to RCS-RDS in previous mails (from 2016 to 2021 according to my notes).

**Comment by ihatethecloud** [comment]
Statically speaking, the bigger an ISP is, the more changes has to have hijacked devices connected to the internet. In my honest opinion, blocking an ISP for having bots and viruses and bad people on the network isn't good security practice. If this was the golden rule of good security there wouldn't be any more online services.

The good thing is generally people who use jenkins can work they way around the censoring. I installed jenkins on a box in the cloud(even though I hate it) and then rsynced everything on my server locally.

My ISP is sponsoring mirrors for major linux distributions. I opened a ticket to ask if they are willing to give jenkins a helping hand. Romania is a huge IT hub with a LOT of big names. I will try my luck with [Hostico.ro](https://hostico.ro) too they are also sponsors for OSS. A local mirror for jenkins would be nice.

**Comment by dduportal** [comment]
> My ISP is sponsoring mirrors for major linux distributions. I opened a ticket to ask if they are willing to give jenkins a helping hand. Romania is a huge IT hub with a LOT of big names. I will try my luck with [Hostico.ro](https://hostico.ro/) too they are also sponsors for OSS. A local mirror for jenkins would be nice.

Many thanks, that could be a way to properly resolve your issue and to help many other Jenkins users while avoiding @C-Otto 's system having to suffer from abuses

**Comment by C-Otto** [comment]
It's not a security issue, it's "just" abuse. The golden rule is providing abuse contact information and working on reported cases. This is not what RCS-RDS is doing, though.

In my opinion we don't censor, we block specific networks. I understand that the outcome is very similar, though.

Again, we'd be happy to also serve Romanian users depending on RCS-RDS. In fact, we try to saturate our 20 GBit/sec link with as much useful traffic as possible. Sadly, not all traffic is working towards this goal, hence the open issues and the unresolved abuse ticket with RCS-RDS.

The optimist in me thinks that if you, as a paying customer, ping them again, they'll respond and start working towards a solution.

**Comment by C-Otto** [comment]
PS: I think linux.ro is a public mirror service provided by RCS-RDS. I don't know more than that, though, as they don't seem to have a public website.

**Comment by ihatethecloud** [comment]
Another update here.
I got a call from RDS and rep said steps will be taken to get the ips unblocked by rwth aachen uni also hosting a Jenkins mirror will be taken into account.

**Comment by santoniu** [comment]
As there is no realistic ETA on this issue to be permanently resolved, is it possible to disable ftp.halifax.rwth-aachen.de mirror for customers connecting from RCS-RDS or should I just go wild and install an external HTTP proxy and use it to update my Jenkins installation?

**Comment by ihatethecloud** [comment]
> As there is no realistic ETA on this issue to be permanently resolved, is it possible to disable ftp.halifax.rwth-aachen.de mirror for customers connecting from RCS-RDS or should I just go wild and install an external HTTP proxy and use it to update my Jenkins installation?

You can't disable rwth-aachen.de yourself. When you call `https://updates.jenkins.io/current/update-center.json` it'a 302 which redirects you to rwth-aachen.de. I tried with nginx and my firewall to proxy [ftp.belnet.be](https://ftp.belnet.be/) but I went down the ssl rabbit hole with no result.

**Comment by C-Otto** [comment]
The outstanding issues and questions have been resolved/answered, I just removed the blocks. Feel free to close this issue. Happy downloading!

**Comment by lemeurherve** [comment]
Many thanks @C-Otto!

Keeping this issue open to follow RCS-RDS & hostico mirrors progress (many thanks @ihatethecloud too for starting that!)

**Comment by ihatethecloud** [comment]
Funny. It looks like my ISP is blocking me from accessing ftp.halifax.rwth-aachen.de.

**Comment by C-Otto** [comment]
Sorry, I messed up. I had duplicate blocks and only removed one of the instances. Should be fixed in a few minutes.

**Comment by ihatethecloud** [comment]
Can somebody from Jenkins jump in a mail with me? I found a contact from RCS which I know since a long time but I can't just tell him to send an email to "jenkins-infra@googlegroups.com".

**Comment by lemeurherve** [comment]
Sure, you can use/forward to jenkins-infra-team@googlegroups.com, it's a private mailbox reserved to infra members, I'll take care of it.

**Comment by lemeurherve** [comment]
We received a mail from Hostico to host a new Jenkins mirror in Romania 🎉 

I'll post update on this issue.

**Comment by C-Otto** [comment]
That's great news - let me know when we should block another provider/country :D

**Comment by ihatethecloud** [comment]
Wow. Shoutout to Hostico for being amazing!

**Comment by lemeurherve** [comment]
> Funny. It looks like my ISP is blocking me from accessing ftp.halifax.rwth-aachen.de.

@ihatethecloud are you still blocked?

**Comment by ihatethecloud** [comment]
> > Funny. It looks like my ISP is blocking me from accessing ftp.halifax.rwth-aachen.de.
> 
> > Sorry, I messed up. I had duplicate blocks and only removed one of the instances. Should be fixed in a few minutes.
> 
> @ihatethecloud are you still blocked?

No. Worked right away after @C-Otto 

> Sorry, I messed up. I had duplicate blocks and only removed one of the instances. Should be fixed in a few minutes.

**Comment by ihatethecloud** [comment]
> Thanks, closing this issue, I opened #3976 to follow the new mirror from Hostico.
> 
> @ihatethecloud Send us a mail at [jenkins-infra-team@googlegroups.com](mailto:jenkins-infra-team@googlegroups.com) if you think RDS would be willing to host a mirror too.

I send an email to a person who is taking care of mirrors but have yet received a reply. I will keep you posted if I get an answer.

**Comment by KweezyCode** [comment]
any fix?

**Comment by C-Otto** [comment]
RCS&RDS isn't blocked anymore.

**Comment by ihatethecloud** [comment]
> RCS&RDS isn't blocked anymore.

Something is happening though....

```
java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl.connect(Unknown Source)
	at java.base/java.net.SocksSocketImpl.connect(Unknown Source)
	at java.base/java.net.Socket.connect(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at java.base/sun.net.NetworkClient.doConnect(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.openServer(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.openServer(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.New(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.followRedirect0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.followRedirect(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1323)
Caused: java.io.IOException: Failed to load https://updates.jenkins.io/download/plugins/authentication-tokens/1.113.v81215a_241826/authentication-tokens.hpi to /var/jenkins_home/plugins/authentication-tokens.jpi.tmp
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1334)
Caused: java.io.IOException: Failed to download from https://updates.jenkins.io/download/plugins/authentication-tokens/1.113.v81215a_241826/authentication-tokens.hpi (redirected to: https://ftp.halifax.rwth-aachen.de/jenkins/plugins/authentication-tokens/1.113.v81215a_241826/authentication-tokens.hpi)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1368)
	at hudson.model.UpdateCenter$DownloadJob._run(UpdateCenter.java:1925)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:2237)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1899)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:121)
	at java.base/java.lang.Thread.run(Unknown Source)```

**Comment by C-Otto** [comment]
As said before, the issue is resolved. If you have another issue, please reach out to ftp@halifax.rwth-aachen.de. This isn't a Jenkins issue, and it's not THIS Jenkins issue.
